### PR TITLE
SB-66: Add status field on Reservations Form and Calendar

### PR DESCRIPTION
**Pull request by EzeLamar**
Part 3/3

Changes:
* Add status field on Reservation Form.
* Update CRUD reservation adding the status field.
* Update Calendar
  * Add red border style for events with status booked (not paid).
  * Add grey color style for events with status canceled.

image

image

### CODE REVIEW

**Comment by EzeLamar** [comment]
[Add missing fields on  Reservation Form](https://trello.com/c/QBmPHuMl/66-add-missing-fields-on-reservation-form)

**Review by lubualo — CHANGES_REQUESTED** [review]

File: app/components/Reservations/ReservationForm.tsx [hunk 1]
```diff
@@ -36,6 +38,22 @@ const TYPE_VALUES = [
...
+	{
+		key: 'paied',
+		label: 'Pagado',
+	},
+	{
+		key: 'cancelled',
+		label: 'Cancelado',
+	},
+];
+
```
[the remark this answers is not in the document]
Comment (EzeLamar) [reply app/components/Reservations/ReservationForm.tsx:56]: After a meeting, we decided not to use reference constants allocated to the database. Instead, we will restrict by firebase rules which are the valid values we can save. Also, in this PR I will define the key values ('booked', 'paid', 'canceled' ) as constants in a `module.ts` file.

File: app/components/Reservations/ReservationForm.tsx [hunk 2]
```diff
@@ -38,6 +43,21 @@ const TYPE_VALUES = [
 	},
 ];
 
+const STATUS_VALUES = [
+	{
+		key: ReservationStatus.Booked,
+		label: 'Reservado',
+	},
+	{
```
Comment (lubualo) [inline app/components/Reservations/ReservationForm.tsx:51]: Great wording in my opinion here!

**Review by lubualo — APPROVED** [review]
Way to go my friend!

**Pull request merged**